src/native/corehost/apphost/apphost.windows.cpp

```diff
@@ -108,6 +103,23 @@ namespace
                 }
             }
         }
+        else if (error_code == StatusCode::BundleExtractionFailure)
+        {
+            pal::string_t line;
+            pal::stringstream_t ss(g_buffered_errors);
+            while (std::getline(ss, line, _X('\n'))) {
+                if (starts_with(line, _X("Bundle header version compatibility check failed."), true))
```

Comment:
Have we stopped producing `Bundle header version compatibility check failed.` in the newer fxr versions?

Reply:
Ah, I see the message always means that a newer runtime is needed.

Reply:
I am just thinking about scenario when the current framework does not understand versioning at all - is there a chance it would just produce "corrupted bundle` error and we will not open a dialog?

Reply:
Right 

> We do have a check for really old hostfxr, like 3.1 which didn't support the new bundles, in which case the apphost will fail with a good error and I think it should show a dialog about needing to install newer version.

So it is either old fxr, which we know about and will pop a dialog, or it is a new version-aware fxr and will produce this message, which will now trigger the dialog.
Either way there will be a dialog.

Reply:
The change is intentionally very targetted - so basically only 6.0 single-file running on 5.0 hostfxr will hit this. And only when the header version doesn't match.

For 7.0 we should review this hole dialog popup functionality and improve on it
* Better mechanism through which the various components can communicate errors
* Possibly show the dialog in more scenarios